src/dataset.py

```diff
@@ -98,43 +99,43 @@ def _augment_with_salary(record: Dict[str, str]) -> Dict[str, str]:
     return context
 
 
-def badge_lookup(badge: str) -> str:
-    try:
-        r = BADGE_DATASET.get(badge)
-        if not r:
-            html = "<p><b>No officer found for this badge number</b></p>"
-        else:
-            context = _augment_with_salary(r)
-            html = render_template("officer.j2", **context)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
+def _build_sql_query(queries: List[Tuple]) -> (str, Tuple):
+    # (column, operator, value)
+    query_list = []
+    query_tuple = ()
+    for query in queries:
+        if query[2]:
+            query_list.append(" ".join(query[:2]) + " ?")
+            query_tuple += (query[2],)
+    return " AND ".join(query_list), query_tuple
```

Comment:
I believe this will also work, and makes the code a bit clearer. Additionally, we want to be sure we wrap the `LIKE` parameters in wildcards so that users can make partial searches. I also think that the parameters can be any sequence, and don't need to be a tuple specifically.

```suggestion
    query_statements = []
    parameters = []
    for field, operator, value in queries:
        if not value:
            continue
        if operator == "LIKE":
            # Wrap query parameters in wildcards
            value = f"%{value}%"
        query_statements.append(f"{field} {operator} ?")
        parameters.append(value)
    return " AND ".join(query_statements), parameters
```